Update Node to v6.11.2

sbeca opened the pull request:
This PR updates Node to v6.11.2. It also updates all dependencies to the latest versions that are possible while maintaining support for Node v6.11.2.

I tried updating to Node v7.10.0 but it appears that SWIG doesn't yet support it as I got a bunch of compile errors so decided that v6.11.2 was a good compromise.

sbeca committed:
Update Node to v6.10.3

sbeca commented:
Tests are currently failing and I'm not exactly sure why. I'll have to look into it some more.

zuker commented:
@sbeca thanks for PR!
It seems that test should use weaker equality assertions (which don't check prototypes) with node v6
see: 
https://travis-ci.org/zuker/box2d-native/builds/236650425#L7583
https://travis-ci.org/zuker/box2d-native/builds/236650425#L7559

zuker commented on demo/index.html:
```diff
@@ -111,19 +110,14 @@
-        })();
+
+        let fps = 60;
+        setInterval(function() {
```
`setInteval` is not good idea for animation loop in browsers
https://stackoverflow.com/questions/38709923/why-is-requestanimationframe-better-than-setinterval-or-settimeout

sbeca commented:
Hi @zuker, I have updated the code so that all tests pass now and have switched back to using requestAnimationFrame. I have also rebased to remove the commit noise so this PR is ready for review again.

sbeca commented:
Hi @zuker, any update on this pull request?

sbeca committed:
Update Node to v6.11.2

endel commented:
Apparently, this pull request on SWIG fixes this: https://github.com/swig/swig/pull/968
